### PR TITLE
refactor(store): rename to sqlite store and introduce retention policies interface

### PR DESCRIPTION
**Pull request by LNSD**
Before doing deeper improvements in the "message uniqueness" logic, I have refactored the `WakuMessageStore` implementation:

* Rename to `SqliteStore`. Keep the previous imports to keep the number of files in the present PR small.
* Move all implementation files under `sqlite_store` module.
* Extract `isSqliteOnly` logic into the `MessageRetentionPolicy` type/interface.
* Add Time based and Capacity based message retention policies.

### CODE REVIEW

File: waku/v2/node/storage/message/sqlite_store/retention_policy_time.nim [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  let now = getNanosecondTime(getTime().toUnixFloat()) 
+  let retentionTimestamp = now - p.retentionTime.nanoseconds
+  let thresholdTimestamp = retentionTimestamp - p.retentionTime.nanoseconds div 10
+
+  if thresholdTimestamp <= oldestReceiverTimestamp: 
+    return ok()
+
+  let res = db.deleteMessagesOlderThanTimestamp(ts=retentionTimestamp)
+  if res.isErr(): 
+    return err("failed to delete oldes messages: " & res.error())
```
Comment (rymnc) [inline waku/v2/node/storage/message/sqlite_store/retention_policy_time.nim:43]:
```suggestion
    return err("failed to delete oldest messages: " & res.error())
```

File: waku/v2/node/storage/message/sqlite_store/retention_policy_capacity.nim [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{.push raises: [Defect].}
```
Comment (rymnc) [inline waku/v2/node/storage/message/sqlite_store/retention_policy_capacity.nim:1]: This retention policy would be applicable to all types of stores that we have in the future, right?
Would it be beneficial to have this one directory up?

File: waku/v2/node/storage/message/sqlite_store/retention_policy_time.nim [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{.push raises: [Defect].}
```
Comment (rymnc) [inline waku/v2/node/storage/message/sqlite_store/retention_policy_time.nim:1]: Same query re: other stores making use of these functions

File: waku/v2/node/storage/message/sqlite_store/retention_policy.nim [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{.push raises: [Defect].}
+
+import
+  stew/results
+import
+  ../../sqlite
+
+
+type RetentionPolicyResult*[T] = Result[T, string]
```
Comment (rymnc) [inline waku/v2/node/storage/message/sqlite_store/retention_policy.nim:9]: Same query re: other stores making use of these functions and types

**Review by rymnc — APPROVED** [review]
LGTM, just a couple questions!

**Comment by LNSD** [comment]
From @rymnc:

> This retention policy would be applicable to all types of stores that we have in the future, right?
Would it be beneficial to have this one directory up?

Note the `execute` method of `MessageRetentionPolicy` interface depends on the `db: SqliteDatabase` parameter. So it is not reusable with any other store implementation as it is.

I'd rather keep it like this because, at this moment, those retention policies are only relevant if tied to the SQLite message store implementation. We could abstract them more, but it is beyond the scope of these changes.

File: waku/v2/node/storage/message/sqlite_store/retention_policy.nim (hunk 4 above)
Comment (LNSD) [reply waku/v2/node/storage/message/sqlite_store/retention_policy.nim:9]: Check my comment: https://github.com/status-im/nwaku/pull/1141#issuecomment-1243721470

File: waku/v2/node/storage/message/sqlite_store/retention_policy_time.nim (hunk 3 above)
Comment (LNSD) [reply waku/v2/node/storage/message/sqlite_store/retention_policy_time.nim:1]: Check my comment: https://github.com/status-im/nwaku/pull/1141#issuecomment-1243721470

File: waku/v2/node/storage/message/sqlite_store/retention_policy_capacity.nim (hunk 2 above)
Comment (LNSD) [reply waku/v2/node/storage/message/sqlite_store/retention_policy_capacity.nim:1]: Check my comment: https://github.com/status-im/nwaku/pull/1141#issuecomment-1243721470

**Review by jm-clius — APPROVED** [review]
Yes, thanks! :)

File: waku/v2/node/storage/message/sqlite_store/retention_policy_capacity.nim (hunk 2 above)
Comment (rymnc) [reply waku/v2/node/storage/message/sqlite_store/retention_policy_capacity.nim:1]: Absolutely. resolving.

**Review by rymnc — APPROVED** [review]

**Pull request merged**